bin/figma2flutter.dart

```diff
@@ -112,16 +134,87 @@ List<TokenTheme> _processTokens(List<TokenTheme> themes) {
   return processor.themes;
 }
 
+/// Loads all the design token info into a single structure resulting in
+/// something that looks like it would if it was all in one file
+///
+/// Creates a single JSON strcture from the driven by the $metadata.json
+/// Add the $metadata.json and $themes.json
+Map<String, dynamic> _arrangeJsonFilesBySection(String inputFileLocation) {
+  // each element of the token set is in this map keyed by the file name ish
+  Map<String, dynamic> mergedTokenSet = {};
+
+  // load the metadata contents and massage the paths to be map paths
+  final metadataContents = json.decode(
+    File('$inputFileLocation/\$metadata.json').readAsStringSync(),
+  );
+  List<dynamic> metadataTokenSetOrder =
+      metadataContents['tokenSetOrder'] as List<dynamic>;
+  //_print('tokenSetOrder is $metadataTokenSetOrder');
+  mergedTokenSet['\$metadata'] = <String, dynamic>{};
+  mergedTokenSet['\$metadata']['tokenSetOrder'] =
+      metadataTokenSetOrder.map((path) => basename(path.toString())).toList();
+
+  // _print('mergedTokenSet has entries: ${mergedTokenSet.keys.toList()}');
+  // _print(
+  //     'mergedTokenSet.\$metadata has entries: ${mergedTokenSet["\$metadata"]}');
+  // _print(
+  //     'mergedTokenSet.\$metadata.tokenSetOrder has entries: ${mergedTokenSet["\$metadata"]["tokenSetOrder"]}');
+
+  // Load the themes and msassage the paths to the  contents to be map paths
+  final themesContents = json.decode(
+    File('$inputFileLocation/\$themes.json').readAsStringSync(),
+  );
+  mergedTokenSet['\$themes'] = themesContents;
+  // loop across the themes  we have to remove pathing
+  for (dynamic oneTheme in (mergedTokenSet['\$themes'] as List<dynamic>)) {
+    // create a new map so we don't have concurrent modificatin
+    Map<String, dynamic> massagedSelectedTokenSets = {};
+    for (var oneSet
+        in (oneTheme['selectedTokenSets'] as Map<String, dynamic>).entries) {
+      massagedSelectedTokenSets[basename(oneSet.key)] = oneSet.value;
+    }
+    oneTheme['selectedTokenSets'] = massagedSelectedTokenSets;
+    //_print('massaged tokens in selectedTokenSets: $massagedSelectedTokenSets');
+  }
+
+  // Iterate across the "tokenSetOrder" in the $metadata file and
+  // load the json files into their own sections in the map
+  for (var onePath in metadataContents['tokenSetOrder'] as List<dynamic>) {
+    var fullPath = '$inputFileLocation/$onePath.json';
+    Map<String, dynamic> contents =
+        jsonDecode(File(fullPath).readAsStringSync()) as Map<String, dynamic>;
+    mergedTokenSet[basename(onePath as String)] = contents;
+    //print('added ${basename(onePath)} sub keys: ${contents.keys}');
+  }
+  return mergedTokenSet;
+}
+
+/// Parses the input from the files in $metadata.json and returns list of resolved tokens
+List<TokenTheme> _parseFromFileSet(String inputFileLocation) {
```

Comment:
Same goes for this method. Lets add test coverage for this.

Reply:
None of your current tests read from files so I wasn't sure how you wanted to arrange that.

Reply:
Left this function where it was.  The single file load should be moved at the same time this one is _parseInputFromFile()